score-vmware/src/main/java/io/cloudslang/content/vmware/services/VmService.java

```diff
@@ -287,15 +291,177 @@
         return results;
     }
 
+    /**
+     * Method used to connect to a data center to clone an existing virtual machine.
+     *
+     * @param httpInputs Object that has all the inputs necessary to made a connection to data center
+     * @param vmInputs   Object that has all the specific inputs necessary to identify the virtual machine that will be cloned
+     * @return Map with String as key and value that contains returnCode of the operation, success message with task id
+     * of the execution or failure message and the exception if there is one
+     * @throws Exception
+     */
+    public Map<String, String> cloneVM(HttpInputs httpInputs, VmInputs vmInputs) throws Exception {
```

Comment:
I would move all this work inside a different class which handles only cloneVM related functionality as the VMService class is getting too big and is doing too much. Would do the same for the others also.